database/src/main/java/database/helper/Validator.java

```diff
@@ -2,40 +2,34 @@
 
 import database.exception.EmptyValueException;
 import database.exception.IncorrectPropertyNameException;
-import database.exception.IncorrectValueTypeException;
 import database.exception.NullPropertyNameOrValueException;
 
 import java.lang.reflect.Field;
 import java.util.List;
 import java.util.Map;
 
 public class Validator {
+    public static final String FILTER_CANNOT_BE_NULL_MESSAGE = "Property name and value cannot be null";
+    public static final String FILTER_CANNOT_BE_EMPTY_MESSAGE = "Value cannot be empty";
+    public static final String INCORRECT_FILTER_NAME_MESSAGE = "Incorrect filter name";
 
-    public static <V> void validateDatabaseFilters(List<Field> declaredFields, Map<String, V> filters)
-            throws NullPropertyNameOrValueException, EmptyValueException, IncorrectPropertyNameException,
-            IncorrectValueTypeException {
-
-        for (Map.Entry<String, V> filter : filters.entrySet()) {
+    public static void validateDatabaseFilters(List<Field> declaredFields, Map<String, List<String>> filters)
+            throws NullPropertyNameOrValueException, EmptyValueException, IncorrectPropertyNameException {
+        for (Map.Entry<String, List<String>> filter : filters.entrySet()) {
             String propertyName = filter.getKey();
-            V expectedValue = filter.getValue();
+            List<String> expectedValues = filter.getValue();
 
-            if (expectedValue == null || propertyName == null) {
-                throw new NullPropertyNameOrValueException("Property name and value cannot be null.");
+            if (expectedValues == null || propertyName == null) {
+                throw new NullPropertyNameOrValueException(FILTER_CANNOT_BE_NULL_MESSAGE);
             }
-            if (expectedValue.equals("")) {
-                throw new EmptyValueException("Value cannot be empty.");
+            if (expectedValues.isEmpty() || expectedValues.stream().anyMatch(String::isBlank)) {
+                throw new EmptyValueException(FILTER_CANNOT_BE_EMPTY_MESSAGE);
             }
 
-            Field matchingField = declaredFields.stream()
+            declaredFields.stream()
                     .filter(field -> field.getName().equals(propertyName))
-                    .findFirst()
-                    .orElseThrow(() -> new IncorrectPropertyNameException("Incorrect property name: " + propertyName));
-
-            if (!matchingField.getType().isInstance(expectedValue)) {
-                throw new IncorrectValueTypeException("Incorrect value type for filter: " + propertyName
-                        + ". Expected: " + matchingField.getType().getName() + ", but got: " +
-                        expectedValue.getClass().getName());
-            }
+                    .findAny()
+                    .orElseThrow(() -> new IncorrectPropertyNameException(INCORRECT_FILTER_NAME_MESSAGE + ": " + propertyName));
```

Comment:
Too long line, wrap it to another line please.

Reply:
Fixed.